src/VirtoCommerce.CustomerReviews.ExperienceApi/Middleware/EvalVendorRatingMiddleware.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using PipelineNet.Middleware;
+using VirtoCommerce.CustomerReviews.Core.Models;
+using VirtoCommerce.CustomerReviews.Core.Services;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.XDigitalCatalog;
+using VirtoCommerce.XDigitalCatalog.Queries;
+
+namespace VirtoCommerce.CustomerReviews.ExperienceApi.Middleware;
+
+public class EvalVendorRatingMiddleware : IAsyncMiddleware<SearchProductResponse>
+{
+    private readonly IMapper _mapper;
+    private readonly IRatingService _ratingService;
+
+    public EvalVendorRatingMiddleware(IMapper mapper, IRatingService ratingService)
+    {
+        _mapper = mapper;
+        _ratingService = ratingService;
+    }
+    
+    public virtual async Task Run(SearchProductResponse parameter, Func<SearchProductResponse, Task> next)
+    {
+        if (parameter == null)
+        {
+            throw new ArgumentNullException(nameof(parameter));
+        }
+
+        var query = parameter.Query;
+        if (query == null)
+        {
+            throw new OperationCanceledException("Query must be set");
```

Comment:
ArgumentException

Reply:
I follow types used in other middlewares